### PR TITLE
Hello World Python Draft

### PR DESCRIPTION
**Pull request by garrettgu10**

### CODE REVIEW

File: content/workers/tutorials/hello-world-python/index.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+## Using the `env` Parameter
+
+In addition to the `request` parameter, the `env` parameter is also passed to
+the Python `fetch` handler and can be used to access
+[environment variables] (/workers/configuration/environment-variables),
+[secrets](/workers/configuration/secrets),and
+[bindings](/workers/configuration/bindings).
+
+For instance, let's try setting an using an environment variable in a Python
```
Comment (dom96) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
For instance, let's try setting and using an environment variable in a Python
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+```toml
+name = "hello-python-worker"
+main = "entry.py"
+compatibility_date = "2024-01-29"
+
+[vars]
+API_HOST = "example.com"
+```
+
+Now, let's update our `fetch` handler to make use of the new `API_HOSt`
```
Comment (dom96) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
Now, let's update our `fetch` handler to make use of the new `API_HOST`
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    return Response.new(env.API_HOST)
+```
+
+Now, to make use of our `wrangler.toml` file, we can launch the worker using
+simply:
+
+```sh
+wrangler dev
+```
+
```
Comment (dom96) [inline content/workers/tutorials/hello-world-python/index.md]: Might be worth showing a `curl` here too and what it returns, just to tie this all together (like you've done in the section above).

**Review by dom96 — APPROVED** [review]
Really nice and very thorough! Thank you for writing this.

File: content/workers/tutorials/hello-world-python/index.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    --request POST \
+    --data '{"name": "Python"}' http://localhost:8787
+# Hello, Python!
+```
+
+## Using the `env` Parameter
+
+In addition to the `request` parameter, the `env` parameter is also passed to
+the Python `fetch` handler and can be used to access
+[environment variables] (/workers/configuration/environment-variables),
```
Comment (kflansburg) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
[environment variables](/workers/configuration/environment-variables),
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+from js import Response
+
+def fetch(request):
+    return Response.new("Hello World!")
+```
+
+Now, we can run the worker locally using Wrangler.
+
+```sh
+wrangler dev entry.py
```
Comment (KianNH) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
$ wrangler dev entry.py
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+async def fetch(request):
+    name = (await request.json()).name
+    return Response.new(hello(name))
+```
+
+Once we edit the `entry.py`, Wrangler should automatically restart the local
+development server. Now, if we send a POST request with the appropriate body,
+we should get a personalized message.
+
+```sh
```
Comment (KianNH) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
```bash
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+async def fetch(request, env):
+    return Response.new(env.API_HOST)
+```
+
+Now, to make use of our `wrangler.toml` file, we can launch the worker using
+simply:
+
+```sh
+wrangler dev
```
Comment (KianNH) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
$ wrangler dev
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    return Response.new(str(arr))
+```
+
+## Summary and Related Resources
+
+We hope that this small guide demonstrates the full-featured support for Python
+within Cloudflare workers.
+
+If you want to get started building your own projects, review the existing list
+of [Quickstart templates](https://developers.cloudflare.com/workers/get-started/quickstarts/).
```
Comment (KianNH) [inline content/workers/tutorials/hello-world-python/index.md]:
```suggestion
of [Quickstart templates](/workers/get-started/quickstarts/). 
```

File: content/workers/tutorials/hello-world-python/index.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
+---
+updated: TODO
```
Comment (hoodmane) [inline content/workers/tutorials/hello-world-python/index.md]: To fix CI: 
```suggestion
```

**Comment by garrettgu10** [comment]
closed in favor of https://github.com/cloudflare/cloudflare-docs/pull/12887

**Pull request closed without merging**